src/index.js

```diff
@@ -102,6 +109,25 @@ export default async function loader(content, sourceMap, meta) {
       processOptions
     );
   } catch (error) {
+    // Check postcss versions to avoid using PostCSS 7.
+    // For caching reasons, we use the readFileSync and existsSync functions from the context,
+    // not the functions from the `fs` module.
+    if (
+      !hasExplicitDependencyOnPostCSS &&
+      isFileExists(PACKAGE_JSON_PATH, this.fs.existsSync) &&
```

Comment:
Run `postcssFactory().version.startsWith("7.")` before `isFileExists` to avoid unnecessary file check for PostCSS 8 users

Reply:
I also changed it to call `statSync(...).isFile()` instead of `existsSync()`, because `existsSync()` is not available in this version of Webpack.